trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-for-report-helper.ts

```diff
@@ -12,68 +17,86 @@ const feeRecordStatusToDisplayStatus: Record<FeeRecordStatus, FeeRecordDisplaySt
-    getPropertyWithFeeRecordIndexAsKey(feeRecordItems, 'totalReportedPayments'),
-  );
-  const paymentsReceivedDataSortValueMap = getKeyToCurrencyAndAmountSortValueMap(getPropertyWithFeeRecordIndexAsKey(feeRecordItems, 'paymentsReceived'));
-  const totalPaymentsReceivedDataSortValueMap = getKeyToCurrencyAndAmountSortValueMap(
-    getPropertyWithFeeRecordIndexAsKey(feeRecordItems, 'totalPaymentsReceived'),
-  );
+): FeeRecordPaymentGroupViewModelItem[] => {
+  const totalReportedPayments = mapFeeRecordPaymentGroupItemsToTotalReportedPayments(feeRecordPaymentGroups);
+  const totalPaymentsReceived = mapFeeRecordPaymentGroupItemsToTotalPaymentsReceived(feeRecordPaymentGroups);
+
+  return feeRecordPaymentGroups.map((feeRecordPaymentGroup, index) => {
+    const { status, feeRecords, paymentsReceived } = feeRecordPaymentGroup;
 
-  return feeRecordItems.map((feeRecordItem, feeRecordIndex) => {
-    const checkboxId: PremiumPaymentsTableCheckboxId = `feeRecordId-${feeRecordItem.id}-reportedPaymentsCurrency-${feeRecordItem.reportedPayments.currency}-status-${feeRecordItem.status}`;
+    const displayStatus = feeRecordStatusToDisplayStatus[status];
+
+    const checkboxId = mapFeeRecordItemsToCheckboxId(feeRecords, status);
     const isChecked = isCheckboxChecked(checkboxId);
 
+    const feeRecordViewModelItems = mapFeeRecordItemsToFeeRecordViewModelItems(feeRecords);
+    const paymentViewModelItems = mapPaymentItemsToPaymentViewModelItems(paymentsReceived);
+
     return {
-      id: feeRecordItem.id,
-      facilityId: feeRecordItem.facilityId,
-      exporter: feeRecordItem.exporter,
-      reportedFees: {
-        formattedCurrencyAndAmount: getFormattedCurrencyAndAmount(feeRecordItem.reportedFees),
-        dataSortValue: reportedFeesDataSortValueMap[feeRecordIndex],
-      },
-      reportedPayments: {
-        formattedCurrencyAndAmount: getFormattedCurrencyAndAmount(feeRecordItem.reportedPayments),
-        dataSortValue: reportedPaymentsDataSortValueMap[feeRecordIndex],
-      },
-      totalReportedPayments: {
-        formattedCurrencyAndAmount: getFormattedCurrencyAndAmount(feeRecordItem.totalReportedPayments),
-        dataSortValue: totalReportedPaymentsDataSortValueMap[feeRecordIndex],
-      },
-      paymentsReceived: {
-        formattedCurrencyAndAmount: feeRecordItem.paymentsReceived ? getFormattedCurrencyAndAmount(feeRecordItem.paymentsReceived) : undefined,
-        dataSortValue: paymentsReceivedDataSortValueMap[feeRecordIndex],
-      },
-      totalPaymentsReceived: {
-        formattedCurrencyAndAmount: feeRecordItem.totalPaymentsReceived ? getFormattedCurrencyAndAmount(feeRecordItem.totalPaymentsReceived) : undefined,
-        dataSortValue: totalPaymentsReceivedDataSortValueMap[feeRecordIndex],
-      },
-      status: feeRecordItem.status,
-      displayStatus: feeRecordStatusToDisplayStatus[feeRecordItem.status],
+      feeRecords: feeRecordViewModelItems,
```

Comment:
Are we doing any sorting on these before displaying e.g. within a group sorting by facility ID?

Reply:
I had thought about doing that, but delayed that decision due to wanting to get the table to work first. I do think it probably makes sense to order each group by the facility ID though as that is the first column, so I will add that in

Reply:
Following the meeting, I will instead make this sort by reported payments - I have added tests for this as well